plugins/spammer/spammer.go

```diff
@@ -37,12 +38,45 @@ func doSpam(shutdownSignal <-chan struct{}) {
 	}
 
 	if !tangle.IsNodeSyncedWithThreshold() {
+		// log.Infof("worker idle because: !tangle.IsNodeSyncedWithThreshold()")
+		randomSleep()
 		return
 	}
 
+	if !tangle.IsNodeSynced() {
+		// log.Infof("worker idle because: !tangle.IsNodeSynced()")
+		randomSleep()
+		return
+	}
+
+	if peering.Manager().ConnectedPeerCount() == 0 {
+		// log.Infof("worker idle because: peering.Manager().ConnectedPeerCount() == 0")
+		randomSleep()
+		return
+	}
+
+	if maxCPUUsage > 0.0 {
+		cpuUsage, err := CPUUsage()
+		if err == nil {
+			// log.Infof("cpuUsage %.2f\n", cpuUsage)
+			if cpuUsage > maxCPUUsage {
+				// log.Infof("worker idle with cpuUsage %.2f > %.2f", cpuUsage, maxCPUUsage)
+				randomSleep()
+				return
+			}
+		} else { // else cpu usage detection not supported (Windows?)
+			log.Infof("Error in CPUUsage. %s", err)
+		}
+	}
+
+	CPUUsageAdjust(+1)
+	defer CPUUsageAdjust(-1)
+
 	timeStart := time.Now()
 	tips, _, err := tipselection.SelectTips(depth, nil)
 	if err != nil {
+		log.Infof("worker idle because SelectTips err: %s", err)
+		randomSleep()
```

Comment:
Why do we even need these sleeps at all? Just abort the `doSpam` without the sleeps, because it is triggered by a ticker anyway.